src/dice/ffi/metall_internal.hpp

```diff
@@ -5,11 +5,42 @@
 #include <metall/metall.hpp>
 
 namespace dice::metall_ffi::internal {
-    /**
+#if METALL_VERSION >= 2800
+     /**
      * @brief The metall manager type used internally.
      *      This object type is whats actually behind the opaque ::metall_manager * in the interface
      */
-    using metall_manager = metall::manager;
+     using metall_manager = metall::manager;
+#else
+
+     /**
+      * @brief The metall manager type used internally.
+      *      This object type is whats actually behind the opaque ::metall_manager * in the interface.
+      *      This is a wrapper that contains the read_only flag for older versions that do not support read_only() yet.
+      */
+     struct metall_manager : metall::manager {
+          bool read_only_;
+
+          template<typename T>
+          using allocator_type = typename metall::manager::template allocator_type<T>;
+
+          metall_manager(metall::create_only_t op, char const *path) : metall::manager{op, path},
+                                                                       read_only_{false} {
+          }
+
+          metall_manager(metall::open_only_t op, char const *path) : metall::manager{op, path},
+                                                                     read_only_{false} {
+          }
+
+          metall_manager(metall::open_read_only_t op, char const *path) : metall::manager{op, path},
```

Comment:
I would also add that constructor:

`basic_manager(create_only_t, const char *base_path,
                const size_type capacity)`

Reply:
doesn't work correctly at the moment, so no point

Reply:
Hmm, we probably won't test it with older versions and with the new version it works. OK, I agree.